assets/js/components/KeyMetrics/ConversionReportingNotificationCTAWidget.js

```diff
@@ -0,0 +1,120 @@
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useCallback, useState } from '@wordpress/element';
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { useSelect, useDispatch } from 'googlesitekit-data';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import whenActive from '../../util/when-active';
+import ConversionReportingDashboardSubtleNotification from './ConversionReportingDashboardSubtleNotification';
+import { useConversionReportingEventsForTailoredMetrics } from './hooks/useConversionReportingEventsForTailoredMetrics';
+import { MODULES_ANALYTICS_4 } from '../../modules/analytics-4/datastore/constants';
+
+function ConversionReportingNotificationCTAWidget( { Widget, WidgetNull } ) {
+	const [ isSaving, setIsSaving ] = useState( false );
+	const isUserInputCompleted = useSelect( ( select ) =>
+		select( CORE_USER ).isUserInputCompleted()
+	);
+
+	const userInputSettings = useSelect( ( select ) =>
+		select( CORE_USER ).getUserInputSettings()
+	);
+	const keyMetricSettings = useSelect( ( select ) =>
+		select( CORE_USER ).getKeyMetricsSettings()
+	);
+	const hasUserPickedMetrics = useSelect( ( select ) =>
+		select( CORE_USER ).getUserPickedMetrics()
+	);
+
+	const purpose = userInputSettings?.purpose?.values?.[ 0 ];
+	const haveConversionReportingEventsForTailoredMetrics =
+		useConversionReportingEventsForTailoredMetrics( purpose );
+
+	// Initial callout is surfaced to the users with tailored metrics, if detectedEvents setting
+	// has a conversion event associated with the ACR key metrics matching the current site purpose answer.
+	// If new ACR key metrics that can be added are found using haveConversionReportingEventsForTailoredMetrics,
+	// and have not been already included, which is determined by includeConversionTailoredMetrics setting, callout banner should be displayed.
+	const shouldShowInitialCalloutForTailoredMetrics =
+		! hasUserPickedMetrics?.length &&
+		isUserInputCompleted &&
+		haveConversionReportingEventsForTailoredMetrics &&
+		! keyMetricSettings?.includeConversionTailoredMetrics;
+
+	const { setKeyMetricsSetting, saveKeyMetricsSettings } =
+		useDispatch( CORE_USER );
+	const { dismissNewConversionReportingEvents } =
+		useDispatch( MODULES_ANALYTICS_4 );
+
+	const handleCTAClick = useCallback( () => {
+		if ( shouldShowInitialCalloutForTailoredMetrics ) {
+			setIsSaving( true );
+			setKeyMetricsSetting( 'includeConversionTailoredMetrics', true );
+			saveKeyMetricsSettings( {
+				widgetSlugs: undefined,
+			} );
+			setIsSaving( false );
+		}
+
+		dismissNewConversionReportingEvents();
+	}, [
+		setKeyMetricsSetting,
+		saveKeyMetricsSettings,
+		dismissNewConversionReportingEvents,
+		shouldShowInitialCalloutForTailoredMetrics,
+	] );
+
+	const onDismiss = useCallback( () => {
+		dismissNewConversionReportingEvents();
+	}, [ dismissNewConversionReportingEvents ] );
```

Comment:
This callback seems redundant, no? Let's remove it.

Reply:
Agree, removed